### ISSUE TITLE
Clicking the 'activate' link after registering for an account does not actually activate the account

### ISSUE THREAD

**Issue by Woseseltops** [Signbank/Global-signbank#701]
Pinpointed the problem to this line of code:

https://github.com/Signbank/Global-signbank/blob/master/signbank/registration/views.py#L52

**Comment by Woseseltops** [comment]
Okay, this is added, but the ASL functionality this was needed for strangely still does not work. 

After many hours of bug hunting I have concluded that there is a difference between the global and ASL Signbank in that newly registered ASL users get access to the main dataset automatically directly after registering (even before activating the account), while for the Global Signbank you have no permissions until somebody actively grants them to you. 

It seems like somebody (me?) created this ASL behavior on purpose in the past, but I cannot find any code that provides this main dataset permission. Any ideas @susanodd and @vanlummelhuizen ?

**Comment by ocrasborn** [comment]
Hi @Woseseltops , much of it sounds like it should be: since the ASL SB only has one dataset, neither new users nor the system managers should be bothered with anything like permissions for the one and only dataset. We decided to keep the datasets in there, as they might well find purposes for datasets later, like having students work on their own secluded environment with a subset of the data or whatever. So, the dataset feature should be dormant for them, but re-activatable at some point.

**Comment by vanlummelhuizen** [comment]
@Woseseltops No idea at this point. `git log -L` could reveal who did what.

**Comment by Woseseltops** [comment]
Hey @ocrasborn , this was the original functionality:

> 1 User registers. If the Signbank installation uses datasets, the user can request access to one or more datasets automatically.
2a If the user requested a dataset, the dataset owner will get an email, and can go to the manage dataset page.
2b If not, s/he will show up in the list of users without dataset (both as a view and as a recurring email), where s/he can be given access to the default dataset.

Let me know if you think that 2b should be changed into: 's/he should be given access to the main dataset automatically'.

**Comment by ocrasborn** [comment]
@Woseseltops : let's not change that, no, for Global Signbank. But for ASL Signbank, I do think that is what we want. That is, after the new user's account has been confirmed by the system admin (Julie) -- right? Otherwise anyone who registers can automatically access the ASL dataset, which I don't think is what they wanted, or is it?

**Comment by Woseseltops** [comment]
> That is, after the new user's account has been confirmed by the system admin (Julie) -- right?

Yes, and in the quote above it is proposed to do this 'confirmation' in the form of giving it access to the main dataset. So there are three things about an account that determine permissions:

* Active vs inactive. Inactive is used for accounts that didn't confirm their email or have been disabled (former colleagues).
* Groups: determines what kind of actions you are allowed to perform.
* Individual dataset permissions: determines which part of the data you are allowed to see.

The fact that the ASL Signbank doesn't use the dataset functionality in their interface doesn't mean we cannot use it under the hood to restrict fresh accounts from access to the data :) .

**Comment by Woseseltops** [comment]
Another morning of bug hunting, and I'm still stuck on the problem. Why do fresh ASL account get access to the main dataset automatically? I've pinpointed the problem to this chunk, where the actual Django user object is created. 

https://github.com/Signbank/Global-signbank/blob/master/signbank/registration/models.py#L93

If you stop the code right there, and request the datasets for this new user with `get_objects_for_user(new_user, 'view_dataset', Dataset)` in the Global Signbank, you get `<QuerySet []>`, as expected. When you do the exact same thing on the ASL server, you get: `<QuerySet [<Dataset: ASL>]>`. So Guardian adds the dataset permission directly as the user is created.

**Comment by Woseseltops** [comment]
FINALLY!

![image](https://media.tenor.com/images/ad29c5da8391f140af1e52a074aeece7/tenor.gif)

A few hours, and two accidental discoveries later, it works:
* I managed to pinpoint the problem even further, to the line where the user was given the 'guest' group
* Discovered a StackOverflow answer talking about group permissions for Django guardian.

Turns out, in the ASL database, the guest group had permission to the ASL dataset! In the database, so that's why I couldn't find it in the code anywhere.
